server/sonar-server/src/main/java/org/sonar/server/computation/task/projectanalysis/step/NewSizeMeasuresStep.java

```diff
@@ -93,45 +101,68 @@ public void execute() {
         .withVariationSupport(periodsHolder)
         .buildFor(ImmutableList.of(duplicationFormula)))
-      .visit(treeRootHolder.getRoot());
+          .visit(treeRootHolder.getRoot());
   }
 
-  private static class NewDuplicationCounter implements Counter<NewDuplicationCounter> {
+  private static class NewSizeCounter implements Counter<NewSizeCounter> {
+    private final MeasureRepository measureRepository;
     private final DuplicationRepository duplicationRepository;
     private final ScmInfoRepository scmInfoRepository;
+    private final Metric nclocDataMetric;
     private final IntVariationValue.Array newLines = IntVariationValue.newArray();
-    private final IntVariationValue.Array newBlocks = IntVariationValue.newArray();
+    private final IntVariationValue.Array newDuplicatedLines = IntVariationValue.newArray();
+    private final IntVariationValue.Array newDuplicatedBlocks = IntVariationValue.newArray();
 
-    private NewDuplicationCounter(DuplicationRepository duplicationRepository, ScmInfoRepository scmInfoRepository) {
+    private NewSizeCounter(MeasureRepository measureRepository, MetricRepository metricRepository, DuplicationRepository duplicationRepository,
+      ScmInfoRepository scmInfoRepository) {
+      this.measureRepository = measureRepository;
       this.duplicationRepository = duplicationRepository;
       this.scmInfoRepository = scmInfoRepository;
+      this.nclocDataMetric = metricRepository.getByKey(NCLOC_DATA_KEY);
     }
 
     @Override
-    public void aggregate(NewDuplicationCounter counter) {
+    public void aggregate(NewSizeCounter counter) {
+      this.newDuplicatedLines.incrementAll(counter.newDuplicatedLines);
+      this.newDuplicatedBlocks.incrementAll(counter.newDuplicatedBlocks);
       this.newLines.incrementAll(counter.newLines);
-      this.newBlocks.incrementAll(counter.newBlocks);
     }
 
     @Override
     public void initialize(CounterInitializationContext context) {
       Component leaf = context.getLeaf();
+      context.getPeriods().forEach(period -> newLines.increment(period, 0));
       if (leaf.getType() != Component.Type.FILE) {
         context.getPeriods().forEach(period -> {
-          newLines.increment(period, 0);
-          newBlocks.increment(period, 0);
+          newDuplicatedLines.increment(period, 0);
+          newDuplicatedBlocks.increment(period, 0);
         });
         return;
       }
-      Iterable<Duplication> duplications = duplicationRepository.getDuplications(leaf);
       Optional<ScmInfo> scmInfo = scmInfoRepository.getScmInfo(leaf);
-
       if (!scmInfo.isPresent()) {
         return;
       }
 
-      DuplicationCounters duplicationCounters = new DuplicationCounters(scmInfo.get(), context.getPeriods());
+      initNewLines(leaf, scmInfo.get(), context.getPeriods());
+      initNewDuplicated(leaf, scmInfo.get(), context.getPeriods());
+    }
+
+    private void initNewLines(Component component, ScmInfo scmInfo, List<Period> periods) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused method parameter "periods". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1172)